extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcContextPropagationConfig.java

```diff
@@ -1,16 +1,19 @@
 package io.quarkus.arc.deployment;
 
+import java.util.Optional;
+
 import io.quarkus.runtime.annotations.ConfigGroup;
 import io.quarkus.runtime.annotations.ConfigItem;
 
 @ConfigGroup
 public class ArcContextPropagationConfig {
 
     /**
-     * If set to true and SmallRye Context Propagation extension is present then enable the context propagation for CDI
-     * contexts.
+     * Support for context propagation will be enabled if the {@code quarkus-smallrye-context-propagation} extension is present,
+     * and this value is {@code true}. If this value is unset then the support will be enabled unless the {@code quarkus-vertx}
+     * extension is present and the {@code io.quarkus.vertx.runtime.VertxCurrentContextFactory} is registered.
```

Comment:
I don't think users will know when `VertxCurrentContextFactory` is registered.

Reply:
Hm, I don't think regular users will ever touch this config property. It was a temp solution, more like a safety switch. And the relevant [config property](https://github.com/quarkusio/quarkus/blob/main/extensions/vertx/deployment/src/main/java/io/quarkus/vertx/deployment/VertxBuildConfig.java#L15) is not visible in the docs.

Reply:
Maybe we could add `generateDocumentation = false` here as well?

Reply:
Sounds reasonable

Reply:
So this is unset by default, if I understand correctly? Which means that if `quarkus-vertx` is present, context propagation will be disabled even if `quarkus-smallrye-context-propagation` is present?

Reply:
> So this is unset by default, if I understand correctly? Which means that if `quarkus-vertx` is present, context propagation will be disabled even if `quarkus-smallrye-context-propagation` is present?

Yes. That's  the _new_ behavior. And context propagation for _CDI_ will be disabled...

Reply:
> And context propagation for _CDI_ will be disabled...

Right. Good to highlight that!

In any case, with the Vert.x duplicated context based ArC context factory (ETOOMANYCONTEXTS), CDI context propagation is not necessary on the event loop, but will stop working on worker threads (or basically whenever the thread-local fallback is required), right?

Reply:
> will stop working on worker threads

Nope, worker threads have the duplicated context propagated.

> whenever the thread-local fallback is required

Yes, e.g. if you create your own thread.

Reply:
Okay, so I admit I don't have the full context here, but this sounds like a pretty big deal to me? Especially if the config option were undocumented, as suggested above in this thread.

Reply:
> but this sounds like a pretty big deal to me?

I should have been more clear -  if you create your own thread then context propagation does not work either. You'd have to use the CP API, i.e. `org.eclipse.microprofile.context.ManagedExecutor`. The question is how many users use the CP API directly? 

The problem is that although we got rid of threadlocals the `ArcContextProvider` does a lot of useless stuff in 99% of use cases.

> Especially if the config option was undocumented, as suggested above in this thread.

Undocumented is the config property `quarkus.vertx.customize-arc-context`, i.e. in 99.99% the `VertxCurrentContextFactory` is registered.

Reply:
> I know that we for example recommend ManagedExecutor as the thread pool that people should use, see e.g. https://github.com/quarkusio/quarkus/discussions/28862. I'm not sure how many extensions use it directly, but I know SmallRye Fault Tolerance does :-)

I'm not 100% sure but the default blocking executor in quarkus is backed by Vert.x worker threads so in theory it should as well (duplicated context is there). 

> I'm not sure half-disabling CDI context propagation is a good first step

I get your point. Let's see what others think... 

BTW it's not "half-disabling" MP CP CDI context propagation, it's _"fully-disabling"_ MP CP CDI and also note the in many cases CDI is the only thing that gets propagated by MP CP in quarkus.

Reply:
I called it "half-disabling" because the CDI request context will sometimes be propagated (through other means than MP ConProp) and sometimes won't be. That is _weird_.